app/assets/scripts/utils/map-utils.js

```diff
@@ -18,26 +39,55 @@ module.exports.getProjectCentroids = function (projects, features) {
     let locations = get(project, 'location');
     if (locations && Array.isArray(locations)) {
       locations.forEach(function (location) {
-        // TODO look at district or marker to see if there's more granular data
-        const region = location.district.governorate;
-        regions[region] = regions[region] || [];
-        regions[region].push(project);
+        const region = location.district;
+        let metadata;
+        region.district.length && region.district.toLowerCase() !== 'all'
+          ? metadata = {type: 'district', fallback: region.governorate}
+          : metadata = {type: 'governorate', fallback: false};
+        const regionId = region[metadata.type];
+        regions[regionId] = regions[regionId] || Object.assign(metadata, {regions: []});
+        regions[regionId].regions.push(project);
       });
     }
   });
 
+  let {districts, egy2} = features;
```

Comment:
Let's use `features[DISTRICT]` and `features[GOVERNORATE]` since that's how it's referred to up top.